packages/turf-helpers/index.ts

```diff
@@ -207,30 +207,39 @@ export function feature<
  * var geometry = turf.geometry(type, coordinates);
  * // => geometry
  */
-export function geometry(
-  type:
+export function geometry<
+  T extends
     | "Point"
     | "LineString"
     | "Polygon"
     | "MultiPoint"
     | "MultiLineString"
     | "MultiPolygon",
+>(
+  type: T,
   coordinates: any[],
```

Comment:
In a simliar fashion, we could narrow this `any[]` based on `T`, but that would be a breaking change, we can file it against the 8.0 milestone. I suppose you could also just make the case that we should do this in 7.x, because it is explicitly a bug in the caller if they aren't passing in the correctly shaped coordinates. My only concern with that is some amount of weirdness with `number[]` vs `[number, number]` issues, which might crop up depending on how consumers wrote their own types.